kotest-assertions/kotest-assertions-json/src/jvmMain/kotlin/io/kotest/assertions/json/JsonMatchers.kt

```diff
@@ -1,81 +1,208 @@
 package io.kotest.assertions.json
 
-import com.fasterxml.jackson.databind.ObjectMapper
-import com.fasterxml.jackson.module.kotlin.registerKotlinModule
 import com.jayway.jsonpath.JsonPath
 import com.jayway.jsonpath.PathNotFoundException
+import io.kotest.assertions.failure
 import io.kotest.matchers.Matcher
 import io.kotest.matchers.MatcherResult
 import io.kotest.matchers.should
 import io.kotest.matchers.shouldNot
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.contract
+import kotlin.reflect.typeOf
+
+@OptIn(ExperimentalContracts::class)
+infix fun Json?.shouldMatchJson(json: Json?) {
+    contract {
+        returns() implies (this@shouldMatchJson != null)
+    }
+
+    this should matchJson(json)
+}
+
+@OptIn(ExperimentalContracts::class)
+infix fun Json?.shouldNotMatchJson(json: Json?) {
+    contract {
+        returns() implies (this@shouldNotMatchJson != null)
+    }
+
+    this shouldNot matchJson(json)
+}
+
+fun matchJson(json: Json?) = object : Matcher<Json?> {
+
+    override fun test(value: Json?): MatcherResult {
+        val actualJson = value?.let { mapper.readTree(it) }
+        val expectedJson = json?.let { mapper.readTree(it) }
+
+        return MatcherResult(
+            actualJson == expectedJson,
+            "expected: $expectedJson but was: $actualJson",
+            "expected not to match with: $expectedJson but match: $actualJson"
+        )
+    }
+}
+
+infix fun Json.shouldMatchJsonResource(resource: String) = this should matchJsonResource(resource)
+infix fun Json.shouldNotMatchJsonResource(resource: String) = this shouldNot matchJsonResource(resource)
+
+fun matchJsonResource(resource: String) = object : Matcher<Json> {
+
+    override fun test(value: Json): MatcherResult {
+        val actualJson = mapper.readTree(value)
+        val expectedJson = mapper.readTree(this.javaClass.getResourceAsStream(resource))
+
+        return MatcherResult(
+            actualJson == expectedJson,
+            "expected: $expectedJson but was: $actualJson",
+            "expected not to match with: $expectedJson but match: $actualJson"
```

Comment:
Maybe this sentence could improve a little bit?
What about
```suggestion
            "expected not to match with: $expectedJson but did."
```

This way we don't print the same sentence twice. Unless there's something I'm not seeing

Reply:
I'm not the author here. Git annotate shows the following commit message:

> Update Json matcher error messages (#1106) (#1112)

Your proposal seems fine to me but maybe I'm not seeing something here too...